### PR TITLE
nim: 0.19.4 -> 0.20.0

### PR DESCRIPTION
**Pull request by royneary**
###### Motivation for this change
Update to the latest release of Nim.

What I did:
- ~~makeWrapper not needed anymore~~
- link sqlite (needed for the the stdlib sqlite module)
- update nodejs to 11.x
- use default phase order (fixes linker errors during test phase)
- substitution in ./test/async/tioselectors.nim not needed anymore
- two more tests need to be disabled
- LD=$CC workaround not needed anymore
- update mosdepth for it to compile with nim 0.20.0

nrpl compiles with the new version.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by mmahut — APPROVED** [review]
- [X] reviewed the diff and commit messages
- [X] made sure ofBorg evals
- [X] run nix-review without any failures
- [X] run and tested the binaries

**Comment by veprbl** [comment]
@GrahamcOfBorg build nim mosdepth nrpl

**Comment by veprbl** [comment]
cc @ehmry @peterhoeg

**Comment by royneary** [comment]
Some tests [lack support for aarch64](https://github.com/nim-lang/Nim/issues/11463). I disabled them for the architecture.

**Comment by peterhoeg** [comment]
Nice cleanups as well! Could you please run the relevant ```pre``` and ```post``` hooks from the various phases as well?

Also, are you sure that ```tzdata``` isn't needed a proper buildInput?

**Comment by royneary** [comment]
I'm triggering the pre and post hooks now (and don't override patchPhase anymore).

I think tzdata was only added as a dependency to fix the path ``/usr/share/zoneinfo`` in a test. So it's only needed during the checkPhase and not as a buildInput.

**Comment by veprbl** [comment]
> I think tzdata was only added as a dependency to fix the path `/usr/share/zoneinfo` in the test case `./tests/stdlib/ttimes.nim`. So it's only needed during the checkPhase and not as a buildInput.

You mean this?
https://github.com/NixOS/nixpkgs/blob/567a05f6ab1ae35ffc076959aed188e27a74b190/pkgs/development/compilers/nim/default.nix#L69

In that case we could drop it from `nativeBuildInputs`. Or otherwise move to `checkInputs`.

**Comment by royneary** [comment]
Oh, I didn't know ``checkInputs``. The same applies for ``nodejs-slim-11_x`` and ``coreutils`` then.

**Comment by veprbl** [comment]
@royneary Could you please squash the nim-related commits?
Also might want to fix your commit author email. You could use "royneary@users.noreply.github.com" if you don't want to share your personal address.

**Comment by royneary** [comment]
@veprbl I squashed the commits. Are you suggesting to change my email address because github wasn't able to verify my GPG signatures? I fixed that.

**Comment by veprbl** [comment]
@royneary Thanks! I was mostly worried about the commit attribution to your username.

There is an eval error now. I guess you missing some parenthesis.

**Comment by royneary** [comment]
@veprbl fixed. It was because in master someone removed variable that was then unused but now used again :-)

**Comment by veprbl** [comment]
@GrahamcOfBorg build nim mosdepth nrpl

File: pkgs/development/compilers/nim/default.nix [hunk 1]
```diff
@@ -48,35 +47,53 @@ stdenv.mkDerivation rec {
...
     ./koch tools -d:release
-  '';
 
-  installPhase = ''
-    install -Dt $out/bin bin/* koch
-    ./koch install $out
-    mv $out/nim/bin/* $out/bin/ && rmdir $out/nim/bin
-    mv $out/nim/*     $out/     && rmdir $out/nim
-    wrapProgram $out/bin/nim \
-      --suffix PATH : ${lib.makeBinPath [ stdenv.cc ]}
```
Comment (veprbl) [inline pkgs/development/compilers/nim/default.nix]: This wrapper is still needed, I think.

Comment (veprbl) [reply pkgs/development/compilers/nim/default.nix]: Does "nim compile" work for you in pure environment?

Comment (royneary) [reply pkgs/development/compilers/nim/default.nix]: No, it doesnt, you are right. I fixed it.

**Comment by veprbl** [comment]
@GrahamcOfBorg build nim mosdepth nrpl

**Review by veprbl — APPROVED** [review]

**Pull request merged**